src/service/github.spec.ts

```diff
@@ -0,0 +1,59 @@
+import { afterEach, describe, expect, it, vi } from "vitest";
+import { setupServer } from "msw/node";
+import { createGitHubDeployment, createJobSummary } from "./github";
+import { getOctokit } from "@actions/github";
+import { mockGithubDeployments } from "../test/mocks";
+import { getTestConfig } from "../test/test-utils";
+import mockfs from "mock-fs";
+import { readFile } from "fs/promises";
+
+afterEach(() => {
+	mockfs.restore();
+});
+
+describe("github", () => {
+	it("Calls createGitHubDeployment successfully", async () => {
+		const githubUser = "mock-user";
+		const githubRepoName = "wrangler-action";
+		const server = setupServer(
+			...mockGithubDeployments({ githubUser, githubRepoName }).handlers,
```

Comment:
Is there anything we should assert on the mock?

Reply:
We are asserting on the request payload [here](https://github.com/cloudflare/wrangler-action/pull/325/files#diff-fd928f78f59ad2b55c465896ff5ebab1a82008825bae3460fb791dcf40efd873R16-R27), I suppose we could also check the response payload, but the response schema is [pretty long](https://docs.github.com/en/rest/deployments/deployments?apiVersion=2022-11-28#create-a-deployment), and we're not currently returning any values from [createGithubDeployment](https://github.com/cloudflare/wrangler-action/pull/325/files#diff-79d58800fd43beb67fbbc9bb955038da16724c49838b6f0d17a52a45458cda97R45).

I don't know if there's much value in mocking the response payload for those reasons

Reply:
No, I think this is okay. All we want to do is test that we're reaching out the way we should be. We aren't doing anything with the response, so I don't think we need to test that it comes back a certain way. The request assertion seems good!